### DISCUSSION TITLE
Enable authentication for Milvus after deployment

### DISCUSSION THREAD

**Discussion by redserpent7** [milvus-io/milvus#26988] [Q&A and General discussion] [answered]
Hi,

I was wondering if there is a way to enable authentication for a Milvus cluster deployed on AWS EKS after the cluster was already deployed and running.

The Milvus documentation here suggests editing the values.yaml file and add a few lines to it to enable authentication: https://milvus.io/docs/authenticate.md#helm

I am not an expert on helm so I am not sure how can I get my cluster to only take the authentication value and don't update anything else in fear that something might break

is there a way to use helm's --set command to enable the configuration in command line directly?

**Comment by yhmo** [comment]
To change configurations of an existing cluster, just modify the values.yaml:
```
extraConfigFiles:
  user.yaml: |+
    common:
      security:
        authorizationEnabled: true
```

Then run the command `helm upgrade` to take effect.
The milvus helm package page mentioned it: https://artifacthub.io/packages/helm/milvus/milvus#breaking-changes

**Reply by redserpent7** [reply, marked as answer]
Ok I figured it out, instead of modifying the existing values.yaml file, I created a new one with just the needed authentication config, then upgraded the cluster with the command

`helm upgrade --reuse-values -f new_values.yaml milvus milvus/milvus`

This was the only way that worked for me without helm rebuilding the entire cluster from scratch
